src/sql/src/plan/statement/show.rs

```diff
@@ -1035,17 +1037,34 @@ fn humanize_sql_for_show_create(
                         }
                     });
                 }
-                CreateSourceConnection::Kafka { .. }
-                | CreateSourceConnection::LoadGenerator { .. } => {}
+
+                CreateSourceConnection::LoadGenerator { .. } => {
+                    // Multi-output load generator sources only support `FOR ALL
+                    // TABLES`.
+                    if !curr_references.is_empty() {
```

Comment:
This is really scaring me. Where does this limitation come from? We need to fix it instead of adding yet another special case. We already added one as part of the subsource dependency inversion and for all we know we might be sitting on a serious subsource handling bug.

Reply:
there are two types of load generator sources––single- and multi-output. Single-output load generators don't have any references and don't require and don't support any `FOR...` references. Multi-output sources _only_ support `FOR ALL TABLES`.

Reply:
> Multi-output sources only support FOR ALL TABLES.

Yes but as mentioned in the subsource inversion PR nothing that I'm aware of should impose that restriction so the fact that it exists tells me there is a bug somewhere. It could be a bug just in load gen sources or it could be a more general bug. Until we look into it we won't know

Reply:
The generator that writes the data assumes that all of the collections it intends to write to exist––it's a deficiency in how they're built. e.g. there's no sense of expressing which subset of the collections to render in [the `auction` load generator](https://github.com/MaterializeInc/materialize/blob/240a3e6a93dfc76dcf6cc8c6c56d933f4c09e629/src/storage/src/source/generator/auction.rs#L73-L180); it only renders them all.

Reply:
But the source render trait does not require that you produce exactly the collections you were asked to. It should be safe to produce more than you were asked to and the ingestion framework should drop them on the floor

Reply:
My comment was in regards to this comment:

> It could be a bug just in load gen sources or it could be a more general bug. Until we look into it we won't know

I am proposing that this is a deficiency in load gen sources.